### ISSUE TITLE
Cannot register or login

### ISSUE THREAD

**Issue by oo00spy00oo** [ellie/atuin#504]
I got the message when tried to register

```
❯ atuin register -u <USERNAME> -p <PASSWORD>
Please enter email: <EMAIL>
Error: failed to register user

Location:
    /private/tmp/atuin-20220606-67432-rpf7h8/atuin-0.10.0/atuin-client/src/api_client.rs:61:9
```

**Comment by conradludgate** [comment]
Are you trying to use the public sync server, or a private one?

**Comment by conradludgate** [comment]
This makes me realise though that we have bad error messages in this path, I'll submit a patch later

**Comment by ellie** [comment]
Hey @oo00spy00oo, if you try again with the latest Atuin (v11):

1. does the error still occur?
2. if so, could you please share the error message? It should be more helpful now :)

**Comment by mzfr** [comment]
@conradludgate @ellie I tried both v11 and v12 and I'm getting different error on both of them

I used the `docker-compose.yml` on my raspi to start the atuin server and it worked without any issue. 

Then on raspi(atuin v11) I got the following:

```bash
$ atuin register -u myuser -e mail@example.com -p password
Error: failed to register user

Location:
    /build/atuin/src/atuin-11.0.0/atuin-client/src/api_client.rs:61:9
```

On my local system(atuin v12) I got:

```bash
$ atuin register -u myuser -e mail@example.com -p password
Error: error decoding response body: expected value at line 1 column 1

Caused by:
    expected value at line 1 column 1

Location:
    /home/runner/work/atuin/atuin/atuin-client/src/api_client.rs:62:21
```

In both cases the only thing that I have modified inside `~/.config/atuin/config.toml` is the `sync_address`

```
sync_address = "http://MY_IP_ADDRESS:8888/"
```

Inside docker, atuin is running v12

### Fix

The error is because of `/` in the end of the URL. If sync_address ends with `/` it will throw the above-mentioned errors but if it doesn't then all good.

* Error(s)

```
sync_address = "http://MY_IP_ADDRESS:8888/"
```

* No error(s)

```
sync_address = "http://MY_IP_ADDRESS:8888"
```
__Notice no / at the end of the URL__

**Comment by conradludgate** [comment]
Thanks for the report and the investigation. That should be an easy fix by parsing the Url early and using the proper push methods instead of string manipulation

**Comment by conradludgate** [comment]
dupe of #389 and being fixed in #979
